src/cpp_generator.cpp

```diff
@@ -84,8 +86,11 @@ void cpp_generator::fmap(const std::function<std::string(std::string)>& f) { imp
 std::string cpp_generator::generate_point_op(const operation& op,
                                              const std::vector<std::string>& args)
 {
-    auto v = op.to_value();
-    return interpolate_string(op.attributes()["point_op"].to<std::string>(),
+    auto v          = op.to_value();
+    auto attributes = op.attributes();
+    if(not attributes.contains("point_op"))
+        MIGRAPHX_THROW("op is missing point_op attribute: " + op.name());
+    return interpolate_string(attributes["point_op"].to<std::string>(),
```

Comment:
The interpolate_string called here is:

```
template <class F>
inline std::string
interpolate_string(const std::string& input, F f, std::string start = "${", std::string end = "}")
{
    std::string result = "";
    result.reserve(input.size());
    auto it = input.begin();
    while(it != input.end())
    {
        auto next_start = std::search(it, input.end(), start.begin(), start.end());
        auto next_end   = std::search(next_start, input.end(), end.begin(), end.end());
        result.append(it, next_start);
        if(next_start == input.end())
            break;
        auto r = f(next_start + start.size(), next_end - end.size() + 1);
        result.append(r.begin(), r.end());
        it = next_end + 1;
    }
    return result;
}

```
I think the next_end-end.size + 1 in the line ```auto r = f(next_start + start.size(), next_end - end.size() + 1);``` does not make sense. The reason is next_end is the start of "end". Move the iterator forward does not make sense.

Reply:
> Move the iterator forward does not make sense.

Its not moving the iterator forward, its returning the distance.

Reply:
Then what does it mean for "next_end - end.size() + 1"? My understanding is it is the location end.length() characters before the "end".

Reply:
You right it is moving the iterator by `end.size() + 1`. Let me investigate that more.

Reply:
So I fixed this and added more tests which found a few more bugs, which I fixed.